src/features/settings/components/BookmarkSettings/BookmarkSettings.tsx

```diff
@@ -60,17 +61,72 @@ const BookmarkItem = ({ item, appendRef = false }: BookmarkItemProps) => {
 }
 
 export const BookmarkSettings = () => {
-  const { userBookmarks } = useBookmarks()
+  const inputFile = useRef<HTMLInputElement | null>(null)
+  const { bookmarkPost, userBookmarks } = useBookmarks()
+  const [downloadURL, setDownloadURL] = useState('')
+  const downloadFileName = 'hackertabBookmarks'
+
+  useEffect(() => {
+    // Generating URL for exporting bookmarks
+    const bookmarkData = new Blob([JSON.stringify(userBookmarks, null, 2)], {
+      type: 'application/json',
+    })
+    setDownloadURL(URL.createObjectURL(bookmarkData))
+  }, [userBookmarks])
+
+  const ImportBookmarks = () => {
+    inputFile.current?.click()
+  }
+
+  const handleFileChange = (event: any) => {
+    if (event.target.files[0]) {
+      const reader = new FileReader()
+      reader.onload = () => {
+        const importData = JSON.parse(reader.result as string)
+        const uniqueImportData = importData.filter(
+          (data: BookmarkedPost) =>
+            !userBookmarks.some((bm) => bm.title === data.title && bm.url === data.url)
+        )
+        uniqueImportData.forEach((data: BookmarkedPost) => {
+          bookmarkPost(data)
+        })
```

Comment:
Instead of bookmarking each item individually, you can use a method called `initState` within the `useBookmarks` store. This method should enable you to set all items at once.